sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/xds/handler/CdsHandler.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Based on dubbo-xds/src/main/java/org/apache/dubbo/registry/xds/util/protocol/AbstractProtocol.java
+ * from the Apache dubbo project.
+ */
+
+package io.sermant.implement.service.xds.handler;
+
+import com.google.protobuf.Any;
+import com.google.protobuf.InvalidProtocolBufferException;
+
+import io.envoyproxy.envoy.config.cluster.v3.Cluster;
+import io.envoyproxy.envoy.service.discovery.v3.DiscoveryRequest;
+import io.envoyproxy.envoy.service.discovery.v3.DiscoveryResponse;
+import io.grpc.stub.StreamObserver;
+import io.sermant.implement.service.xds.cache.XdsDataCache;
+import io.sermant.implement.service.xds.client.XdsClient;
+import io.sermant.implement.service.xds.env.XdsConstant;
+import io.sermant.implement.service.xds.utils.XdsProtocolTransformer;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+import java.util.logging.Level;
+
+/**
+ * CdsHandler
+ *
+ * @author daizhenyu
+ * @since 2024-05-13
+ **/
+public class CdsHandler extends XdsHandler {
+    /**
+     * construction method
+     *
+     * @param client xds client
+     * @param resourceType xds resource type
+     */
+    public CdsHandler(XdsClient client, String resourceType) {
+        super(client, resourceType);
```

Comment:
just provide a constructor 'public CdsHandler(XdsClient client)'，resourceType is never changed in CdsHandler.   Same situation in  EdsHandler

Reply:
Create the resourcetype attribute in the parent class. The subclass specifies the attribute by using the construction method.

Reply:
super(client);
this.resourceType = xxx

Reply:
done